File: frontend/src/types/ChatMessage.ts
import type { ChatFile } from "./ChatFile"
import type { ChatMessageType } from "./ChatMessageType"

export type ChatMessage = {
  content: string,
  contentIV: string,
  sentAt: number,
  type: ChatMessageType,
  orderId?: number,
  id?: number,
  senderId?: number,
  files?: ChatFile[],
  chatPartnerId?: number,
}

File: backend/cmd/ofm_backend/api/my_profile/dto/chat_message.go
package dto

import (
	"ofm_backend/cmd/ofm_backend/enum"
	"time"
)

type ChatMessage struct {
	Id        int64                `json:"id"`
	SenderId  int64                `json:"senderId"`
	Content   []byte               `json:"content"`
	ContentIV []byte               `json:"contentIV"`
	SentAt    time.Time            `json:"sentAt"`
	Files     []string             `json:"files"`
	Type      enum.ChatMessageType `json:"type"`
}
